### PR TITLE
tests: nrf_provisioning: Add CoAP tests

### PR DESCRIPTION
**Pull request by juhaylinen**
Add tests for CoAP provisioning client.

### CODE REVIEW

**Comment by SeppoTakalo** [comment]
I think there is now too many tags in the tittle to my liking...
It does not have to reflect exactly the directory structure.

I think `tests: nrf_profisioning` is just fine.
Similarly like `net: nrf_provisioning:` is fine.

File: tests/subsys/net/lib/nrf_provisioning/include/coap_client.h [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/** @file
+ * @brief CoAP client API
+ *
+ * An API for applications to do CoAP requests
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (SeppoTakalo) [inline tests/subsys/net/lib/nrf_provisioning/include/coap_client.h:10]: Cannot use Apache.
Code inside `nrf` repository must use the Nordic-5-clause license.

File: tests/subsys/net/lib/nrf_provisioning/include/coap_client.h [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+	struct coap_block_context send_blk_ctx;
+	struct coap_pending pending;
+	struct coap_client_request *coap_request;
+	struct coap_packet request;
+//	k_tid_t tid;
+//	struct k_thread thread;
+//	struct k_sem coap_client_recv_sem;
+//	atomic_t coap_client_recv_active;
+//
+//	K_THREAD_STACK_MEMBER(coap_thread_stack, CONFIG_COAP_CLIENT_STACK_SIZE);
```
Comment (SeppoTakalo) [inline tests/subsys/net/lib/nrf_provisioning/include/coap_client.h:100]: Drop out the commented fields.

**Review by SeppoTakalo — COMMENTED** [review]
OK otherwise, but please fix the compliance warnings.

**Review by SeppoTakalo — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

**Pull request merged**